docs/configuration.asciidoc

```diff
@@ -0,0 +1,265 @@
+[[configuration]]
+== Configuration
+
+[float]
+=== Gradle configuration
+
+Configure your application at compile time within your application's `build.gradle` file:
+
+[source,groovy]
+----
+// Android app's build.gradle file
+plugins {
+    //...
+    id "co.elastic.apm.android" version "[latest_version]"
+}
+
+elasticApm {
+    // Minimal configuration
+    serverUrl = "https://your.elastic.server"
+
+    // Optional
+    serviceName = "your app name" <1>
+    serviceVersion = "0.0.0" <2>
+    secretToken = "your server auth token" <3>
+}
+----
+<1> Defaults to your `android.defaultConfig.applicationId` value.
+<2> Defaults to your `android.defaultConfig.versionName` value.
+<3> Defaults to null.
+
+All of the values provided in the Gradle configuration can be overridden with the following environment variables:
+
+|===
+|Config |Associated Environment variable
+
+|serviceName
+|`ELASTIC_APM_SERVICE_NAME`
+
+|serviceVersion
+|`ELASTIC_APM_SERVICE_VERSION`
+
+|serverUrl
+|`ELASTIC_APM_SERVER_URL`
+
+|secretToken
+|`ELASTIC_APM_SECRET_TOKEN`
+
+|===
+
+[float]
+=== Runtime configuration
+
+The runtime configuration is provided within your https://developer.android.com/reference/android/app/Application[Application] class at the time of initializing the Elastic agent.
+This configuration overrides any previously-set compile time configuration.
+
+The way it works is by providing your own instance of the `ElasticApmConfiguration` class as shown below
+
+[source,java]
+----
+// Application class
+
+class MyApp extends android.app.Application {
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+        ElasticApmAgent.initialize(this, ElasticApmConfiguration.builder().build());
+    }
+}
+----
+
+[float]
+==== APM server connectivity
+
+The server connectivity parameters can be provided at compile time, either by using the gradle dsl configuration, or by providing the server connectivity related environment variables as mentioned above.
+Later on, when the app is running, the connectivity parameters can be overriden by providing a custom `Connectivity` instance when initializing the Elastic agent.
+
+The `Connectivity` interface provides a set of convenient methods to avoid having to create your own implementation in case you need to define a custom connectivity.
+
+Once you've created your `Connectivity` instance, you can set it into the agent's initialization as show below:
+
+[source,java]
+----
+class MyApp extends android.app.Application {
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+        Connectivity myCustomConnectivity = Connectivity.custom(/*params*/);
+        ElasticApmAgent.initialize(this, myCustomConnectivity);
+
+        // Optionally if you also define a custom configuration:
+        // ElasticApmAgent.initialize(this, ElasticApmConfiguration.builder().build(), myCustomConnectivity);
+    }
+}
+----
+
+[float]
+==== Application id configuration
```

Comment:
Should ID be capitalized here?

Reply:
Now that you mention it, it makes sense to capitalize it yeah! I just pushed the changes :)